collector/src/credential.rs

```diff
@@ -29,21 +29,15 @@ impl PrivateCollectorCredential {
         self.token.clone().map(AuthenticationToken::Bearer)
     }
 
-    /// Returns the [`HpkeKeypair`] necessary for decrypting aggregate shares. This errors if the
-    /// configuration references any HPKE parameters that are not supported by the application.
+    /// Returns the [`HpkeKeypair`] necessary for decrypting aggregate shares. This cannot fail
+    /// currently, but returns a `Result` for historical reasons.
     pub fn hpke_keypair(&self) -> Result<HpkeKeypair, Error> {
```

Comment:
I'd say that we should provide a new, infallible version and mark this one deprecated, but I'm not sure what name we'd use for the new version.

Reply:
I gave it an ugly name, `hpke_keypair_infallible()`, and we can iron it out in 0.7.